### PR TITLE
Improve documentation #18

### PR DESCRIPTION
**Pull request by m-mohr**
Attempt to improve the docs, especially the supported options.

Daniel, could you please verify this is correct and complete? Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,10 +38,25 @@ layer.addTo(map);
...
+| buildUrlTemplate             | function      | undefined                                  | For server-side rendering of imagery. See the chapter [using a tiler](#using-a-tiler) and [buildTileUrlTemplate](#buildTileUrlTemplate) for details. |
+| displayPreview               | boolean       | false                                      | Allow to display the asset with role `thumbnail` or the link with relation type `preview`. The previews are usually not covering the full extents and as such may be placed incorrectly on the map. |
+| displayOverview              | boolean       | true                                       | Allow to display the asset with role `overview` or `visual`. |
+| debugLevel                   | integer       | 0                                          | The higher the value the more debugging messages will be logged to the console. `0` to disable logging. |
+| latLngBounds / bounds / bbox | array<number> | undefined                                  | For STAC Assets only: The bounding box of the asset |
```
Comment (m-mohr) [inline README.md:50]:
```suggestion
| latLngBounds / bounds / bbox | array\<number> | undefined                                  | For STAC Assets only: The bounding box of the asset |
```

And we likely need to tell what the array should contain for all these options...

File: README.md [hunk 2]
```diff
@@ -39,10 +38,25 @@ layer.addTo(map);
...
+| baseUrl                      | string        | `href` of the `self` link in the STAC data | The base URL for relative links. Should be provided if the STAC data has no self link and links are not absolute. |
+| buildUrlTemplate             | function      | undefined                                  | For server-side rendering of imagery. See the chapter [using a tiler](#using-a-tiler) and [buildTileUrlTemplate](#buildTileUrlTemplate) for details. |
+| displayPreview               | boolean       | false                                      | Allow to display the asset with role `thumbnail` or the link with relation type `preview`. The previews are usually not covering the full extents and as such may be placed incorrectly on the map. |
+| displayOverview              | boolean       | true                                       | Allow to display the asset with role `overview` or `visual`. |
```
Comment (DanielJDufour) [inline README.md:48]: The code currently doesn't require displayOverview to be true to display assets with the "visual" key:
https://github.com/stac-utils/stac-layer/blob/169883c0c49b5c17d1a1aaa863eee2f2dd925f19/src/index.js#L361

I'd recommend updating this to "Allow to display the asset with role `overview`".  Or we can update the code from:
```if (addedImagery === false && hasAsset(assets, "visual")) {```
to
```if (addedImagery === false && displayOverview && hasAsset(assets, "visual")) {```

Let me know your preference :-)

File: README.md (hunk 1 above)
Comment (DanielJDufour) [reply README.md:50]: Maybe we can update dataType to 

[latLngBounds](https://leafletjs.com/reference.html#latlngbounds) | [bounds](https://leafletjs.com/reference.html#bounds) | [West, South, East, North]

or 

[latLngBounds](https://leafletjs.com/reference.html#latlngbounds) | [bounds](https://leafletjs.com/reference.html#bounds) | [MinLongitude, MinLatitude, MaxLongitude, MaxLatitude]

File: README.md [hunk 3]
```diff
@@ -39,10 +38,25 @@ layer.addTo(map);
...
+| debugLevel                   | integer       | 0                                          | The higher the value the more debugging messages will be logged to the console. `0` to disable logging. |
+| latLngBounds / bounds / bbox | array<number> | undefined                                  | For STAC Assets only: The bounding box of the asset |
+| resolution                   | integer       | 32                                         | Adjust the display resolution, a power of two such as 32, 64, 128 or 256. By default the value is set to render quickly, but with limited resolution. Increase this value for better resolution, but slower rendering speed (e.g., 128). |
```
Comment (DanielJDufour) [inline README.md:51]: Looks good!

File: README.md [hunk 4]
```diff
@@ -39,10 +38,25 @@ layer.addTo(map);
...
+| tileUrlTemplate              | string        | undefined                                  | For server-side rendering of imagery. See the chapter [using a tiler](#using-a-tiler) and [tileUrlTemplate](#tileUrlTemplate) for details. |
+| titiler                      | string        | undefined                                  | URL to a titiler instance that can be used to serve tiles instead of rendering the imagery client-side. |
```
Comment (DanielJDufour) [inline README.md:53]: The titiler option was removed with the removal of the non-standard compositing of separated R-G-B bands. 
 People can still use titiler though through the other options:
```js
tileUrlTemplate: "https://titiler.tld/cog/tiles/{z}/{x}/{y}?url={url}"
```

Comment (DanielJDufour) [reply README.md:53]: Some test examples are here: https://github.com/stac-utils/stac-layer/tree/main/src/test/tilers/titiler.  Wondering if we want to include links to the tests here?  Or create separate examples?

**Review by DanielJDufour — CHANGES_REQUESTED** [review]
Left a few comments.

File: README.md (hunk 2 above)
Comment (m-mohr) [reply README.md:48]: As "visual" is not standardized, I see it as a "fallback" for "overview. So I think I'd prefer the option to add the displayOverview check into the if clause.

Comment (DanielJDufour) [reply README.md:48]: @m-mohr , sounds good.  I just updated the code to reflect this: https://github.com/stac-utils/stac-layer/commit/ed80a5eb498d228318adfbf7fc089a6d812e7a57

Comment (m-mohr) [reply README.md:48]: Great, thanks a lot!

File: README.md (hunk 4 above)
Comment (m-mohr) [reply README.md:53]: Interesting, I found it in the code. So are the following occurrences outdated and should be removed?

- https://github.com/stac-utils/stac-layer/blob/1e2ffb7db2c91ee7b37404edf0c715ffd880a3d6/src/test/tilers/titiler/cbers.html#L33
- https://github.com/stac-utils/stac-layer/blob/ed80a5eb498d228318adfbf7fc089a6d812e7a57/src/index.js#L91-L92

Comment (DanielJDufour) [reply README.md:53]: Good catch.  I'll remove those.

Comment (DanielJDufour) [reply README.md:53]: Done with https://github.com/stac-utils/stac-layer/commit/041461c9ec73d93bec2cb25b5fe36b974954c85e

Comment (DanielJDufour) [reply README.md:53]: I'm wondering what you would like me to do with these two files?
https://github.com/stac-utils/stac-layer/blob/main/src/utils/titiler.js and https://github.com/stac-utils/stac-layer/blob/main/src/utils/titiler-layer.js.  They are both not being used, but could be very helpful in the future if you decide to add functionality to this library that specifically requires ti-tiler.

Comment (DanielJDufour) [reply README.md:53]: Or we could even pull these out into a separate NPM package if you like?

Comment (m-mohr) [reply README.md]: Just leave them in for now, as you said they could be useful for some. Maybe we should add a short readme for it?

File: README.md (hunk 1 above)
Comment (m-mohr) [reply README.md]: Done.

**Comment by m-mohr** [comment]
Thanks for the review, I think all points have been addressed. Could you please re-review and merge if this is good to go?

File: README.md (hunk 4 above)
Comment (DanielJDufour) [reply README.md]: sounds good

**Pull request merged**